src/routes/sessions.js

```diff
@@ -62,50 +60,76 @@ module.exports = function (app, opts, dependencies) {
     };
   }
 
+  function getTwoFactorResponse(user) {
+    if (process.env.LIANA_2FA_SECRET  === undefined) {
+      logger.error('Cannot use the two factor authentication because the environment variable "LIANA_2FA_SECRET" is not set.');
+      throw new Error();
+    }
+
+    if (process.env.LIANA_2FA_SECRET.length !== 20) {
+      logger.error('The environment variable "LIANA_2FA_SECRET" must be 20 characters long.');
+      logger.error('You can generate it using this command: `$ openssl rand -hex 10`');
+      throw new Error();
+    }
+
+    if (user.two_factor_authentication_configured) {
+      return { twoFactorAuthenticationEnabled: true };
+    } else {
+      const partialUserSecret = user.partial_user_secret;
+      const userSecret = new UserSecretCreator(partialUserSecret, process.env.LIANA_2FA_SECRET)
+        .perform();
+
+      return {
+        twoFactorAuthenticationEnabled: true,
+        userSecret,
+      };
+    }
+  }
+
   function loginWithPassword(request, response) {
     const renderingId = request.body.renderingId;
     const envSecret = opts.envSecret;
+    const { email, password } = request.body;
 
-    return ipWhitelist.retrieve(envSecret)
-      .then(function () { return new AllowedUsersFinder(renderingId, envSecret).perform(); })
-      .then(function (allowedUsers) {
-        if (allowedUsers.length === 0) {
-          throw new Error(errorMessages.SESSION.NO_USERS);
-        }
+    (async function asyncWrapper() {
```

Comment:
What is the benefit of such code VS `try/catch` syntax?

Reply:
With try/catch you still need to wrap the `await` call to an `async function ...`